### PR TITLE
ci/test: use cargo-deny to check licenses and duplicate dependencies

### PR DESCRIPTION
**Pull request by benesch**
Since Rust crates tend to be tiny, it's very easy to accidentally bring
in a ton of duplicate transitive dependencies. Use cargo-deny to lint
against this happening. When duplicate dependencies are unavoidable,
they can be explicitly allowed in the deny.toml config file.

As an added bonus, cargo-deny will also prevent us from depending on
non-OSS code.

This commit also upgrades us to Rust 1.41, since cargo-deny depends on
features in Rust 1.41. The code changes are the result of changes in
Rust 1.41's rustfmt.

### CODE REVIEW

File: src/billing-demo/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ description = "Microservice demo using Materialize to power a real-time billing
 version = "0.1.0"
 authors = ["Brandon W Maister <bwm@materialize.io>"]
 edition = "2018"
+publish = false
```
Comment (quodlibetor) [inline src/billing-demo/Cargo.toml]: should we add this to the lint script?

File: src/ccsr/tests/client.rs [hunk 2]
```diff
@@ -231,17 +231,14 @@ fn start_server(runtime: &Runtime, status_code: StatusCode, body: &'static str)
...
         let addr = incoming.local_addr();
-        let server = Server::builder(incoming).serve(service::make_service_fn(move |_conn| {
-            async move {
-                Ok::<_, hyper::Error>(service::service_fn(move |_req| {
-                    async move {
-                        Response::builder()
-                            .status(status_code)
-                            .body(Body::from(body))
-                    }
+        let server =
```
Comment (quodlibetor) [inline src/ccsr/tests/client.rs]: are these code changes just because of the rust version upgrade?

**Comment by quodlibetor** [comment]
darn I didn't read the full commit message, nvm about the surprising code changes!

File: rust-toolchain [hunk 3]
```diff
@@ -1 +1 @@
-1.40.0
+1.41.0
```
Comment (quodlibetor) [inline rust-toolchain]: actually I believe this should be 1.41.1: https://blog.rust-lang.org/2020/02/27/Rust-1.41.1.html

Comment (benesch) [reply rust-toolchain]: Gah, thanks. Updating.

File: src/billing-demo/Cargo.toml (hunk 1 above)
Comment (benesch) [reply src/billing-demo/Cargo.toml]: It already is part of the lint script, in a way. Without `publish = false`, the missing license will trigger a `cargo deny` error.

**Pull request merged**